File: src/scraper.js
// /* JS parsing script to parse articles using jssoup*/
// // const fetch = require('node-fetch');
// var JSSoup = require('jssoup').default;

// const scrapeArticle = async (url) => {
//   try {
//     const response = await fetch('https://cors-anywhere.herokuapp.com/' + url, { mode: 'cors' }); //fetching the article using a cors proxy
//     const html = await response.text();
//     const soup = new JSSoup(html); //making soup

//     // Use JSSoup to select the article content element(s) and extract the text
//     const article = soup.findAll("p"); //everything within paragraph tags
//     console.log(article.toString());
//     const articleText = article ? article.text : '';

//     return articleText;
//   } catch (error) {
//     console.error(error);
//     return null;
//   }
// };

// module.exports = scrapeArticle;

File: src/scraper.py
import requests
from bs4 import BeautifulSoup

# Replace this with the URL of the article you want to scrape
url = "https://www.forbes.com/sites/richardnieva/2023/03/10/google-peacetime-ceo-sundar-pichai/?sh=6a508d4c4ab1"

# Send a GET request to the URL and get the HTML content
response = requests.get(url)
html_content = response.content

# Parse the HTML content using BeautifulSoup
soup = BeautifulSoup(html_content, "html.parser")

# Find the main text of the article (usually inside a <div> or <article> tag)
article_text = ""
article_main = soup.find("div", {"class": "article-main"})
if article_main:
    for paragraph in article_main.find_all("p"):
        article_text += paragraph.text + "\n"
else:
    article_main = soup.find("article")
    if article_main:
        for paragraph in article_main.find_all("p"):
            article_text += paragraph.text + "\n"

# Print the scraped article text
print(article_text)